mesonbuild/interpreter/type_checking.py

```diff
@@ -5,12 +5,27 @@
 
 from __future__ import annotations
 import os
+import re
 import typing as T
 
 from .. import compilers
-from ..build import (EnvironmentVariables, EnvInitValueType, CustomTarget, BuildTarget,
-                     CustomTargetIndex, ExtractedObjects, GeneratedList, IncludeDirs,
-                     BothLibraries, SharedLibrary, StaticLibrary, Jar, Executable)
+from ..build import (
+    BothLibraries,
+    BuildTarget,
+    CustomTarget,
```

Comment:
I like it one per line for a couple of reasons:
1. It makes git rebase conflicts *much* easier to sort out, git can oftern figure them out for you, in cases like you add one and master removes 1, which it almost certinaly cant otherwise
2. my editor folds them anyway
3. it makes them easier (personally) to read

Reply:
It makes it harder to read when it scrolls off the screen...

git rebase conflicts don't happen often enough to drastically reduce the readability of a file, they're easy to solve either way, and the way it's done before this PR is extremely idiomatic python which we do everywhere else.

Maybe you could tell your editor to stop producing non-idiomatic behind the scenes edits to your files?

Reply:
> Maybe you could tell your editor to stop producing non-idiomatic behind the scenes edits to your files?

What? Where are you coming up with that?

> git rebase conflicts don't happen often enough to drastically reduce the readability of a file, they're easy to solve either way, and the way it's done before this PR is extremely idiomatic python which we do everywhere else.

git rebase conflicts come up in imports enough that of three series I rebased today *all* of the conflicts were in imports and *all* of them would have been silently handled by git if we sorted them the way I'm proposing.

I also don't know whe're you're getting that it's not idiomiatc? PEP8 doesn't specify either way, and I've seen projects that do it both ways.

Reply:
> What? Where are you coming up with that?

If your editor is doing stuff here for you "anyway" then I'm not sure why that is relevant? Unless it's editing files behind your back as an autosave thing? Why are editors being mentioned here?

> silently handled by git if we sorted them the way I'm proposing.

So then, I ~~eagerly~~ await your PR where you change the three lines almost immediately below this diff hunk to do the same thing. There's cases of two or more imports on the same line, which we could get rid of, right? How about doing a treewide change while we are at it?

> I've seen projects that do it both ways.

So have I, but the ones that do it the way you're suggesting are the ones where I can't read the source code. They also use enforced autoformatters which add commits to your PR to change imports back to their preferred style. They also use:
```python
from a import A
from a import B
from a import C
from b import D
from b import E
from c import F
```

Which is not what you're doing either. (Merge conflicts can occur in the parentheses too, you know!)

A few of them even agree that the code is less readable, but claim that this is preferable to the chance that a merge conflict (these projects do not use rebasing) might occur. They therefore mandate this even for importing exactly two (and no more) imports from a single module.

Reply:
I still think this makes for an unreadable mess, but I'm not going to argue with you about it here. I've reverted the change and gone back to the existing style.